### PR TITLE
Doc: Update download version to 10.1 and fix tar command

### PR DESCRIPTION
**Pull request by acassis**
## Summary
Doc: Update download version to 10.1 and fix tar command
## Impact
Fix issue https://github.com/apache/incubator-nuttx/issues/4562
## Testing
N/A

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: Documentation/quickstart/install.rst [hunk 1]
```diff
@@ -165,22 +165,22 @@ Apache NuttX is actively developed on GitHub. There are two main repositories, `
...
     
        $ mkdir nuttx
        $ cd nuttx
-       $ curl -L https://downloads.apache.org/incubator/nuttx/9.1.0/apache-nuttx-9.1.0-incubating.tar.gz -o nuttx.tar.gz 
-       $ curl -L https://downloads.apache.org/incubator/nuttx/9.1.0/apache-nuttx-apps-9.1.0-incubating.tar.gz -o apps.tar.gz
-       $ tar zxf nuttx.tar.gz
-       $ tar zxf apps.tar.gz
+       $ curl -L https://downloads.apache.org/incubator/nuttx/10.1.0/apache-nuttx-10.1.0-incubating.tar.gz -o nuttx.tar.gz 
+       $ curl -L https://downloads.apache.org/incubator/nuttx/10.1.0/apache-nuttx-apps-10.1.0-incubating.tar.gz -o apps.tar.gz
+       $ tar zxf nuttx.tar.gz --one-top-level=nuttx --strip-components 1
```
Comment (btashton) [inline Documentation/quickstart/install.rst:185]: Is this actually needed for the apache releases or just the GitHub generated ones?

**Review by btashton — CHANGES_REQUESTED** [review]
.

File: Documentation/quickstart/install.rst (hunk 1 above)
Comment (btashton) [reply Documentation/quickstart/install.rst:185]: Also url should be updated to the new style download link

Comment (acassis) [reply Documentation/quickstart/install.rst]: > Is this actually needed for the apache releases or just the GitHub generated ones? The release validation script does not use this, so I am fairly certain this change is not needed here.

Done!

Comment (acassis) [reply Documentation/quickstart/install.rst]: > Also url should be updated to the new style download link

It is better to wait ASF put everything in place first. The download is not working:
curl -L https://www.apache.org/dyn/closer.lua/incubator/nuttx/10.1.0/apache-nuttx-10.1.0-incubating.tar.gz -o nuttx.tar.gz

Do you think we should use the CDN link: https://dlcdn.apache.org/incubator/nuttx/10.1.0/apache-nuttx-10.1.0-incubating.tar.gz ?

**Comment by acassis** [comment]
> .

Done, please review and merge

**Review by btashton — APPROVED** [review]

**Pull request merged**